include/openthread/dns_client.h

```diff
@@ -55,9 +55,64 @@ extern "C" {
  *
  */
 
-#define OT_DNS_DEFAULT_SERVER_PORT 53 ///< The default DNS Server port.
+/**
+ * This enumeration type represents the "Recursion Desired" (RD) flag in an `otDnsQueryConfig`.
+ *
+ */
+typedef enum
+{
+    OT_DNS_FLAG_UNSPECIFIED       = 0, ///< Indicates the flag is not specified.
+    OT_DNS_FLAG_RECURSION_DESIRED = 1, ///< Indicates DNS name server can resolve the query recursively.
+    OT_DNS_FLAG_NO_RECURSION      = 2, ///< Indicates DNS name server can not resolve the query recursively.
+} otDnsRecursionFlag;
+
+/**
+ * This structure represents a DNS query configuration.
+ *
+ * Any of the fields in this structure can be set to zero to indicate that it is not specified.
+ *
+ */
+typedef struct otDnsQueryConfig
+{
+    otSockAddr         mServerSockAddr;  ///< Server address (IPv6 address/port). All zero or zero port for unspecified.
+    uint32_t           mResponseTimeout; ///< Wait time (in msec) to rx response. Zero indicates unspecified value.
```

Comment:
Note that zero is **not** the default value for the timeout. Otherwise default behavior with zero timeout would be that we won't wait at all 😄 

Value zero indicates it is unspecified (i.e. user does not care to give and leave it to stack to choose/use a default)

Reply:
By **default** I meant the starting default values.

It seems unclear what `unspecified value` means just by reading the comment here, even though it's explained later in `otDnsClientSetDefaultConfig`. 

Maybe add a `@sa otDnsClientSetDefaultConfig` ?

Reply:
The goal here in the definition of `otDnsQueryConfig` is to mention what value to use to indicate that the field is unspecified.

Note that what replaces the unspecified field values depends on which API it is being used in (which is then explained in the related method documentation)